pcdsdevices/gon.py

```diff
@@ -236,23 +252,344 @@ class Kappa(BaseInterface, Device):
+        e_eta, e_chi, e_phi = self.k_to_e()
+        return e_eta
+
+    @property
+    def e_chi_coord(self):
+        """Get the elevation (polar) angle, a composition of eta and kappa."""
+        e_eta, e_chi, e_phi = self.k_to_e()
+        return e_chi
+
+    @property
+    def e_phi_coord(self):
+        """Get the sample rotation angle, an offset from phi to keep it."""
+        e_eta, e_chi, e_phi = self.k_to_e()
+        return e_phi
+
+    def k_to_e(self, eta=None, kappa=None, phi=None):
+        """
+        Convert from native kappa coordinates to spherical coordinates.
+
+        If a parameter is left as None, use the live value.
+
+        Parameters
+        ----------
+        eta : number
+            Eta motor position.
+        kappa : number
+            Kappa motor position.
+        phi : number
+            Phi motor position.
+
+        Returns
+        -------
+        coordinates : tuple
+            Spherical coordinates.
+        """
+        if not eta and eta != 0:
+            eta = self.eta_position
+        if not kappa and kappa != 0:
+            kappa = self.kappa_position
+        if not phi and phi != 0:
+            phi = -self.phi_position
+
+        kappa_ang = self.kappa_ang * np.pi / 180
+
+        delta = np.arctan(np.tan(kappa * np.pi / 180 / 2.0) *
+                          np.cos(kappa_ang))
+        e_eta = -eta * np.pi / 180 - delta
+        e_chi = 2.0 * np.arcsin(np.sin(kappa * np.pi / 180 / 2.0) *
+                                np.sin(kappa_ang))
+        e_phi = phi * np.pi / 180 - delta
+        e_eta = e_eta * 180 / np.pi
+        e_chi = e_chi * 180 / np.pi
+        e_phi = e_phi * 180 / np.pi
+        return e_eta, e_chi, e_phi
+
+    def e_to_k(self, e_eta=None, e_chi=None, e_phi=None):
+        """
+        Convert from spherical coordinates to the native kappa coordinates.
+
+        If a parameter is left as None, use the live value.
+
+        Parameters
+        ----------
+        e_eta : number
+        e_chi : number
+        e_phi : number
+
+        Returns
+        -------
+        coordinates : tuple
+            Native kappa coordinates.
+        """
+        if not e_eta and e_eta != 0:
+            e_eta = self.e_eta_coord
+        if not e_chi and e_chi != 0:
+            e_chi = self.e_chi_coord
+        if not e_phi and e_phi != 0:
+            e_phi = self.e_phi_coord
+
+        kappa_ang = self.kappa_ang * np.pi / 180
+
+        delta = np.arcsin(-np.tan(e_chi * np.pi / 180 /
+                                  2.0) / np.tan(kappa_ang))
+        k_eta = -(e_eta * np.pi / 180 - delta)
+        k_kap = 2.0 * np.arcsin(np.sin(e_chi * np.pi /
+                                       180 / 2.0) / np.sin(kappa_ang))
+        k_phi = e_phi * np.pi / 180 - delta
+
+        k_eta = k_eta * 180 / np.pi
+        k_kap = k_kap * 180 / np.pi
+        k_phi = -k_phi * 180 / np.pi
+        return k_eta, k_kap, k_phi
+
+    def mv_e_eta(self, value):
```

Comment:
Methods like this only existed on the old class because they were required to make the virtual motors work- if they aren't being used in forward/inverse they probably aren't needed

Reply:
> I haven't checked the math very carefully against the original, but this looks like a pretty clean transfer

I'm adding some tests against the old values - they seem to work fine, i'll push them soon

Reply:
we're using the `check_motor_step` in those functions, maybe i could include that somewhere in the forward/inverse

Reply:
I think if I add this `check_motor_step` in the `forward` method, it could get pretty annoying for a user, as she/he will have to constantly right `y` for every single move if the step is bigger than max step, but it will be safer probably. 

Or we could leave the current methods of movement as a safer alternative for people - and they'll have the option to move safe or not so safe?

Also, if we end up not using the `check_motor_step` at all, we can still at least print it out for the user so they are aware about the current position vs target position?

Reply:
resolving this because I added the `check_motor_step` as part of the `move` method